addons/sale/models/sale_order.py

```diff
@@ -905,7 +914,11 @@ def _action_confirm(self):
                 if not order.analytic_account_id:
                     order._create_analytic_account()
 
-    def _send_order_confirmation_mail(self):
+    def _send_payment_succeeded_for_order_mail(self):
```

Comment:
Confusing logic mixing order confirmation mail and payment succeeded.
I think the logic should be kept separated one way or another :thinking:.